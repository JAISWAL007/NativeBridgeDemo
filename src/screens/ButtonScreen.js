import React from 'react';
import PropTypes from 'prop-types';
import {SafeAreaView, StyleSheet, Text, View, Alert} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import SwipeButton from 'rn-swipe-button';
import Icon from 'react-native-vector-icons/FontAwesome';

import Strings from '../Constant';
const DiamondIcon = () => <Icon name="diamond" color="#fff" size={20} />;

const ButtonScreen = ({navigation}) => {
  const onPress = () => {
    Alert.alert('Error', Strings.COMING_SOON, [{text: 'Ok'}]);
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.subContainer}>
        <View style={styles.headerStyle}>
          <TouchableOpacity
            testID="back-button"
            onPress={() => navigation.goBack()}>
            <Icon name="angle-left" color="#fff" size={30} />
          </TouchableOpacity>
          <Text style={styles.title}>{Strings.BUTTON_SCREEN}</Text>
          <TouchableOpacity
            testID="user-details"
            onPress={() => navigation.navigate('UserDetails')}>
            <Icon name="home" color="#fff" size={30} />
          </TouchableOpacity>
        </View>
      </View>
      <Text style={[styles.buttonTextStyle, styles.colorText]}>
        {Strings.VARIATION_OF_BUTTON}
      </Text>
      <Text
        style={[styles.buttonTextStyle]}
        onPress={onPress}
        testID="button-1">
        {Strings.PRESS_ME}
      </Text>
      <TouchableOpacity
        style={[styles.buttonWrapperStyle, styles.buttonBackgroundColor]}
        onPress={onPress}
        testID="button-2">
        <Text style={styles.buttonTextStyle}> {Strings.PRESS_ME}</Text>
      </TouchableOpacity>

      <TouchableOpacity onPress={onPress} style={styles.buttonWrapperStyle}>
        <Text
          style={[styles.buttonTextStyle, styles.textColor]}
          testID="button-3">
          {' '}
          {Strings.PRESS_ME}
        </Text>
      </TouchableOpacity>
      <SwipeButton
        railBackgroundColor="black"
        testID="slider-button"
        thumbIconStyles={styles.borderRadius}
        thumbIconComponent={DiamondIcon}
        railBorderColor="#333"
        railFillBorderColor="#33adff"
        railFillBackgroundColor="#333"
        titleColor="#33adff"
        thumbIconBackgroundColor="#33adff"
        title={Strings.SLIDE_ME_TO_CONTINUE}
        railStyles={styles.sliderBackgroundColorStyle}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
  },
  subContainer: {
    flex: 1,
  },
  colorText: {
    color: 'yellow',
  },
  headerStyle: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  item: {
    backgroundColor: '#fff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 20,
  },
  title: {
    fontSize: 28,
    margin: 10,
    color: '#fff',
    textAlign: 'center',
  },
  subTitle: {
    fontSize: 20,
    marginTop: 10,
  },
  buttonWrapperStyle: {
    backgroundColor: '#33adff',
    alignItems: 'center',
    borderRadius: 10,
    justifyContent: 'center',
    margin: 10,
  },
  buttonBackgroundColor: {
    backgroundColor: '#333333',
  },
  buttonTextStyle: {
    fontSize: 16,
    textAlign: 'center',
    margin: 15,
    color: '#66b3ff',
  },
  textColor: {
    color: '#fff',
  },
  sliderBackgroundColorStyle: {
    backgroundColor: '#33300088',
    borderRadius: 10,
  },
  borderRadius: {
    borderRadius: 10,
  },
});

ButtonScreen.prototype = {
  navigation: PropTypes.objectOf({
    navigate: PropTypes.func,
  }).isRequired,
};
export default ButtonScreen;
