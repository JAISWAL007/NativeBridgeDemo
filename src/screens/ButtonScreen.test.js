import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {Alert} from 'react-native';
import ButtonScreen from './ButtonScreen';

describe('Button Screen', () => {
  const mockNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };
  it('Should render button screen', () => {
    const tree = render(<ButtonScreen navigation={mockNavigation} />);
    expect(tree).toMatchSnapshot();
  });

  it('Should able to go back when user click on back arrow button', async () => {
    const {getByTestId} = render(<ButtonScreen navigation={mockNavigation} />);
    const backButton = getByTestId('back-button');
    await fireEvent.press(backButton);
    expect(mockNavigation.goBack).toBeCalledTimes(1);
  });

  it('Should able to navigate to home screen when user click on home button', async () => {
    const {getByTestId} = render(<ButtonScreen navigation={mockNavigation} />);
    const homeButton = getByTestId('user-details');
    await fireEvent.press(homeButton);
    expect(mockNavigation.navigate).toBeCalledTimes(1);
  });

  it('Should able to display popup when user click on onPress 1st button', async () => {
    jest.spyOn(Alert, 'alert');
    const {getByTestId} = render(<ButtonScreen navigation={mockNavigation} />);
    const firstButton = getByTestId('button-1');
    await fireEvent.press(firstButton);
    expect(Alert.alert).toHaveBeenCalledWith('Error', 'coming soon', [
      {text: 'Ok'},
    ]);
  });

  it('Should able to display popup when user click on onPress 2nd button', async () => {
    jest.spyOn(Alert, 'alert');
    const {getByTestId} = render(<ButtonScreen navigation={mockNavigation} />);
    const secondButton = getByTestId('button-2');
    await fireEvent.press(secondButton);
    expect(Alert.alert).toHaveBeenCalledWith('Error', 'coming soon', [
      {text: 'Ok'},
    ]);
  });

  it('Should able to display popup when user click on onPress 3rd button', async () => {
    jest.spyOn(Alert, 'alert');
    const {getByTestId} = render(<ButtonScreen navigation={mockNavigation} />);
    const thirdButton = getByTestId('button-3');
    await fireEvent.press(thirdButton);
    expect(Alert.alert).toHaveBeenCalledWith('Error', 'coming soon', [
      {text: 'Ok'},
    ]);
  });
});
