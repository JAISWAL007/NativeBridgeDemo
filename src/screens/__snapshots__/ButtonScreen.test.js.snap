// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Screen Should render button screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="back-button"
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </RNGestureHandlerButton>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 28,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Button screen
      </Text>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="user-details"
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#66b3ff",
          "fontSize": 16,
          "margin": 15,
          "textAlign": "center",
        },
        Object {
          "color": "yellow",
        },
      ]
    }
  >
    4 variations of button
  </Text>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#66b3ff",
          "fontSize": 16,
          "margin": 15,
          "textAlign": "center",
        },
      ]
    }
    testID="button-1"
  >
    Press me
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="button-2"
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "borderRadius": 10,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#66b3ff",
            "fontSize": 16,
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
         
        Press me
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#33adff",
          "borderRadius": 10,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#66b3ff",
              "fontSize": 16,
              "margin": 15,
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
        testID="button-3"
      >
         
        Press me
      </Text>
    </View>
  </RNGestureHandlerButton>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 50,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 5,
        },
        Object {
          "backgroundColor": "black",
          "borderColor": "#333",
        },
      ]
    }
  >
    <Text
      importantForAccessibility=""
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "color": "#33adff",
            "fontSize": 20,
          },
        ]
      }
    >
      Slide me to continue
    </Text>
  </View>
</RCTSafeAreaView>
`;
